pallets/msa/src/lib.rs

```diff
@@ -853,6 +858,51 @@ pub mod pallet {
 			}
 			Ok(Some(T::WeightInfo::retire_msa(num_deletions)).into())
 		}
+
+		/// create a provider by means of governance approval
+		#[pallet::call_index(11)]
+		#[pallet::weight(T::WeightInfo::create_provider(100_000))]
+		pub fn create_provider_via_governance(
```

Comment:
Since this is a public function for end-users, I think we should name it `request_to_be_a_provider()` or similar.